src/resolver/resolver.zig

```diff
@@ -2846,6 +2846,12 @@ pub const Resolver = struct {
                 const total_length: ?u32 = strings.indexOfChar(original_path, '*');
                 var prefix_parts = [_]string{ abs_base_url, original_path[0 .. total_length orelse original_path.len] };
 
+                // Concatenate the matched text with the suffix from the wildcard path
+                const suffix = std.mem.trimLeft(u8, original_path[total_length orelse original_path.len ..], "*");
+                var matched_text_with_suffix = r.allocator.alloc(u8, matched_text.len + suffix.len) catch unreachable;
+                defer r.allocator.free(matched_text_with_suffix);
+                bun.concat(u8, matched_text_with_suffix, &.{ matched_text, suffix });
```

Comment:
Concatenating these here is necessary as if it is done via the parts array, it would have a backslash in instances where one is not expected (e.g. when the suffix is a file extension like .js).

Not sure if this is the ideal way to concat them though.